core/src/main/scala/org/apache/spark/util/SizeEstimator.scala

```diff
@@ -32,6 +32,16 @@ import org.apache.spark.annotation.DeveloperApi
 import org.apache.spark.util.collection.OpenHashSet
 
 /**
+ * A trait that allows a class to give [[SizeEstimator]] more accurate size estimation.
+ * When a class extends it, [[SizeEstimator]] will query the `estimatedSize` first.
+ * If `estimatedSize` does not return [[None]], [[SizeEstimator]] will use the returned size
+ * as the size of the object. Otherwise, [[SizeEstimator]] will do the estimation work.
+ */
+private[spark] trait SizeEstimation {
+  def estimatedSize: Option[Long]
```

Comment:
I think we do not need to do that.

`SizeEstimator.estimate` (the publish method of `SizeEstimator`) is used at two places. One is memory store and another one is trait `SizeTracker` (a utility trait used to implement collections that need to track estimated size). We do not put `BytesToBytesMap` to memory store, right?